exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -15,97 +15,134 @@
-	// bucketSuffix is the bucket suffix for distribution buckets.
-	bucketSuffix = "bucket"
-	// quantileSuffix is the quantile suffix for summary quantiles.
-	quantileSuffix = "quantile"
 	// countSuffix is the count metric value suffix.
 	countSuffix = "count"
-	// sumOfSquaredDeviation is the sum of squared deviation metric value suffix.
-	sumOfSquaredDeviation = "sum_of_squared_deviation"
-)
-
-var (
-	// infinity bound dimension value is used on all histograms.
-	infinityBoundSFxDimValue = float64ToDimValue(math.Inf(1))
 )
 
 func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config) ([]*splunk.Event, int, error) {
-	ocmds := internaldata.MetricsToOC(data)
 	numDroppedTimeSeries := 0
-	_, numPoints := data.MetricAndDataPointCount()
-	splunkMetrics := make([]*splunk.Event, 0, numPoints)
-	for _, ocmd := range ocmds {
-		var host string
-		if ocmd.Resource != nil {
-			host = ocmd.Resource.Labels[conventions.AttributeHostHostname]
+	var splunkMetrics []*splunk.Event
+	for i := 0; i < data.ResourceMetrics().Len(); i++ {
+		rm := data.ResourceMetrics().At(i)
+
+		host := unknownHostName
+		source := config.Source
+		sourceType := config.SourceType
+		index := config.Index
+		if !rm.Resource().IsNil() {
+			if conventionHost, isSet := rm.Resource().Attributes().Get(conventions.AttributeHostHostname); isSet {
+				host = conventionHost.StringVal()
+			}
+			if sourceSet, isSet := rm.Resource().Attributes().Get(conventions.AttributeServiceName); isSet {
+				source = sourceSet.StringVal()
+			}
+			if sourcetypeSet, isSet := rm.Resource().Attributes().Get(splunk.SourcetypeLabel); isSet {
+				sourceType = sourcetypeSet.StringVal()
+			}
 		}
-		if host == "" {
-			host = unknownHostName
+		rm.InstrumentationLibraryMetrics().Len()
+		commonFields := map[string]interface{}{}
+		if !rm.Resource().IsNil() {
+			rm.Resource().Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+				commonFields[k] = v.StringVal()
+			})
 		}
-		for _, metric := range ocmd.Metrics {
-			for _, timeSeries := range metric.Timeseries {
-				for _, tsPoint := range timeSeries.Points {
-					values, err := mapValues(logger, metric, tsPoint.GetValue())
-					if err != nil {
-						logger.Warn(
-							"Timeseries dropped to unexpected metric type",
-							zap.Any("metric", metric),
-							zap.Any("err", err))
-						numDroppedTimeSeries++
-						continue
+		for ilmi := 0; ilmi < rm.InstrumentationLibraryMetrics().Len(); ilmi++ {
+			ilm := rm.InstrumentationLibraryMetrics().At(ilmi)
+			for tmi := 0; tmi < ilm.Metrics().Len(); tmi++ {
+				tm := ilm.Metrics().At(tmi)
+				switch tm.DataType() {
+				case pdata.MetricDataTypeIntGauge:
+					for gi := 0; gi < tm.IntGauge().DataPoints().Len(); gi++ {
+						fields := copy(commonFields)
+						dataPt := tm.IntGauge().DataPoints().At(gi)
+						populateLabels(fields, dataPt.LabelsMap())
+						fields[fmt.Sprintf("%s:%s", splunkMetricValue, tm.Name())] = dataPt.Value()
```

Comment:
This `string` can be calculated only once before iterating over all points. Also in general is faster `splunkMetricValue  + ":" + tm.Name()`